gbench/common_utils.hpp

```diff
@@ -90,6 +91,50 @@ static inline const char* from_transpose_enum(Transposition t) {
   }
 }
 
+/**
+ * @fn time_event
+ * @brief Times 1 event, and returns the aggregate time.
+ */
+template <typename EventT>
+inline cl_ulong time_event(EventT&);
+// Declared here, defined separately in the specific utils.hpp files
+
+/**
+ * @fn time_events
+ * @brief Times n events, and returns the aggregate time.
+ */
+template <typename EventT>
+inline cl_ulong time_events(std::vector<EventT> es) {
+  cl_ulong total_time = 0;
+  for (auto e : es) {
+    total_time += time_event(e);
+  }
+  return total_time;
+}
+
+template <typename EventT, typename... OtherEvents>
+inline cl_ulong time_events(EventT first_event, OtherEvents... next_events) {
+  return time_events<EventT>(
+      blas::concatenate_vectors(first_event, next_events...));
+}
+
+/**
+ * @fn timef
+ * @brief Calculates the time spent executing the function func
+ * (both overall and event time, returned in nanoseconds in a tuple of double)
+ */
+template <typename F, typename... Args>
```

Comment:
make sure you are using the same coding style as sycl-blas framework. F is not a good name if it reprenset a function change it to function_t